### PR TITLE
Add an API for inventory item atlas

### PR DESCRIPTION
**Pull request by ZzzzzzzSkyward**

### CODE REVIEW

File: scripts/utils/resolveinventoryitemassets.lua [hunk 1]
```diff
@@ -7,6 +7,10 @@ return function(prefab)
   local atlas = GetInventoryItemAtlas(item_tex)
   local localized_name = STRINGS.NAMES[string.upper(prefab)] or prefab
   local prefabData = Prefabs[prefab]
+  
+  local registered_image, registered_altas = GetFoodAtlas(prefab)
+  item_tex = registered_image or item_tex
+  atlas = registered_altas or atlas
 
   if prefabData then
     -- first run we find assets with exact match of prefab name
```
Comment (IMalyugin) [inline scripts/utils/resolveinventoryitemassets.lua:16]: Hmm, so for what reason the original way of providing images doesn't work?
Normally, if you provide Assets section in the inventory item prefab with INV_IMAGE / ATLAS. You said something about using SetPrefabNameOverride. Wouldn't it just work if you also copied the assets for the copied prefabs?

Comment (IMalyugin) [reply scripts/utils/resolveinventoryitemassets.lua:16]: Except winterfeast don't have explicit atlases :)
Hate that part about some of the klei's api

**Comment by IMalyugin** [comment]
You might want to take a look at how Waiter mod adds inventory images, maybe you just need to copy that style? Works with native game: https://steamcommunity.com/sharedfiles/filedetails/?id=381565292

It's not that I don't want to merge your api. It's just I want to be sure we can't add atlas definitions without one, using all native api makes it more resillient. 

I'm looking at another mod issues at the moment anyways

File: scripts/utils/resolveinventoryitemassets.lua (hunk 1 above)
Comment (ZzzzzzzSkyward) [reply scripts/utils/resolveinventoryitemassets.lua:16]: In my case I didn't provide any INV_IMAGE or ATLAS. And you won't find the atlas anywhere before they actually spawn. So either an API or a dynamic parser.

Comment (ZzzzzzzSkyward) [reply scripts/utils/resolveinventoryitemassets.lua:16]: The dynamic parser is what Display Food Values use, spawn and remove to see its real image and atlas from inst.replica.inventoryitem.

**Comment by ZzzzzzzSkyward** [comment]
> You might want to take a look at how Waiter mod adds inventory images, maybe you just need to copy that style? Works with native game: https://steamcommunity.com/sharedfiles/filedetails/?id=381565292
> 
> It's not that I don't want to merge your api. It's just I want to be sure we can't add atlas definitions without one, using all native api makes it more resillient.

Does that mod uses official icon for their custom dishes?

**Comment by IMalyugin** [comment]
Okay, merged the API along with two other fixes, will release the version shortly

**Comment by IMalyugin** [comment]
@ZzzzzzzSkyward you could probably still do without the API:
By reading official icons via:
`local GetInventoryItemAtlas = require "utils/getinventoryitematlas"`

And then applying it via atlases

**Pull request closed without merging**